edx_django_utils/monitoring/internal/code_owner/utils.py

```diff
@@ -103,6 +109,60 @@ def get_code_owner_mappings():
     return _PATH_TO_CODE_OWNER_MAPPINGS
 
 
+def _get_catch_all_code_owner():
+    """
+    If the catch-all module "*" is configured, return the code_owner.
+
+    Returns:
+        (str): code_owner or None if no catch-all configured.
+
+    """
+    try:
+        code_owner = get_code_owner_from_module('*')
+        return code_owner
+    except Exception:  # pylint: disable=broad-except; #pragma: no cover
+        return None
+
+
+def _set_code_owner_attribute_from_module(module):
+    """
+    Updates the code_owner and code_owner_module custom attributes.
+    """
+    set_custom_attribute('code_owner_module', module)
+    code_owner = get_code_owner_from_module(module)
+    if code_owner:
+        set_custom_attribute('code_owner', code_owner)
+        return
+
+    code_owner = _get_catch_all_code_owner()
+    if code_owner:
+        set_custom_attribute('code_owner', code_owner)
+
+
+def set_code_owner_attribute(module):
+    """
+    Decorator to set the code_owner custom attribute for use with celery tasks.
+
+    Celery tasks or other non-web functions do not use middleware, so we need
+        an alternative way to set the code_owner custom attribute.
+
+    Usage::
+
+        @set_code_owner_attribute(__name__)
+        @task()
+        def example_task():
+            ...
+
+    """
+    def actual_decorator(wrapped_function):
+        @wraps(wrapped_function)
+        def new_function(*args, **kwargs):
+            _set_code_owner_attribute_from_module(module)
```

Comment:
**Reviewer Note**: Originally I was using `wrapped_function.__module__`, rather than a module argument, but I feared that cases with compound decorators might report the wrong module.

Reply:
1. Some of the edx-platform tests are failing: https://github.com/edx/edx-platform/pull/25584. I thought this was right, but maybe I'm missing something.
2. Since @task needs to be outer-most, and it relies on the function name,maybe I can trust that `wrapped_function.__module__` would work?  Would removing the function call fix the failures?

Reply:
I tried without the argument and I am getting the same error. I need to start debugging and/or get a second pair of eyes.

Reply:
FYI: I left this as the cleaner alternative, since we now offer an alternative to using the decorator.